### PR TITLE
Develop upstream sync 230731

### PR DESCRIPTION
**Pull request by weihanmines**

### CODE REVIEW

**Comment by i-chaochen** [comment]
Since I did the cover here https://github.com/openxla/xla/pull/4603/files#diff-fc02eb6aea06ad0d72011e9a64da28e8c6fae000a9e0cab9b15b40b4e914af4aR956-R958

could you try to enable triton-softmax flag to have a go, please?
https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/2170/commits/da2cefbc2e15fd87475126e1cf21eddf8a649d2a#diff-04cb485c1774fda54f8346ece0ade9efbcd813235768905b234d596c22660cf6R170

we need to see are any other unit tests affected on that flag.

**Review by i-chaochen — COMMENTED** [review]
Since gpu.graph PR is merged, you should enable xla_graph_level = 1

**Comment by weihanmines** [comment]
> Since gpu.graph PR is merged, you should enable xla_graph_level = 1

turn it on in the latest commit.

**Comment by weihanmines** [comment]
> Since I did the cover here https://github.com/openxla/xla/pull/4603/files#diff-fc02eb6aea06ad0d72011e9a64da28e8c6fae000a9e0cab9b15b40b4e914af4aR956-R958
> 
> could you try to enable triton-softmax flag to have a go, please? [da2cefb#diff-04cb485c1774fda54f8346ece0ade9efbcd813235768905b234d596c22660cf6R170](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/commit/da2cefbc2e15fd87475126e1cf21eddf8a649d2a#diff-04cb485c1774fda54f8346ece0ade9efbcd813235768905b234d596c22660cf6R170)
> 
> we need to see are any other unit tests affected on that flag.

turn it on in the latest commit.

**Comment by weihanmines** [comment]
Jenkins: retest Ubuntu-GPU-single please

File: tensorflow/compiler/xla/service/gpu/BUILD
Comment (i-chaochen) [inline tensorflow/compiler/xla/service/gpu/BUILD:1]: This is added in https://github.com/tensorflow/tensorflow/commit/56f261ba2fad322bacbb69aa1eeaff46b5d2910c we need a ticket to track this test

Comment (i-chaochen) [inline tensorflow/compiler/xla/service/gpu/BUILD:1]: https://github.com/tensorflow/tensorflow/commit/6d990a87f15d6e28397c5d38b56d2545eebbd9ed

This is a hlo and mixed precisions test, we are capble of this. Please create a ticket and we need to support this in TF/XLA

Comment (weihanmines) [reply tensorflow/compiler/xla/service/gpu/BUILD:1]: Will create a ticket for it.

Comment (weihanmines) [reply tensorflow/compiler/xla/service/gpu/BUILD:1]: Sure.

Comment (weihanmines) [reply tensorflow/compiler/xla/service/gpu/BUILD:1]: https://github.com/orgs/ROCmSoftwarePlatform/projects/22?pane=issue&itemId=34779375 
Here is the ticket for the linking issue.

Comment (i-chaochen) [reply tensorflow/compiler/xla/service/gpu/BUILD:1]: I think you disable two tests: ```float_support_test``` and ```hlo_op_profiler_test``` ? 

I can see ```hlo_op_profiler_test``` and ```unary_ops_test``` in the ticket but no ```float_support_test```. 

Also could you put the related upstream PRs to each disable test in the ticket, please?

Comment (weihanmines) [reply tensorflow/compiler/xla/service/gpu/BUILD:1]: `float_support_test` requires NVIDIA Ampere architecture or above. Which upstream do you refer to, rocmsoftwareplatform/tensorflow or tensorflow upstream?

Comment (i-chaochen) [reply tensorflow/compiler/xla/service/gpu/BUILD:1]: When I mentioned upstream, I mean tensorflow upstream. 

```requires-gpu-sm80``` is for NV side, we should capable of suporting this. it's hlo and mixed precision. @jayfurmanek

**Review by jayfurmanek — APPROVED** [review]

File: tensorflow/compiler/xla/service/gpu/BUILD
Comment (i-chaochen) [reply tensorflow/compiler/xla/service/gpu/BUILD:1]: hlo_op_profiler_test is fixed at https://github.com/openxla/xla/pull/4889
